File: src/main/javabak/datagen/ModWorldGenProvider.java
package redcrafter07.processed.datagen;

import net.minecraft.core.HolderLookup;
import net.minecraft.core.RegistrySetBuilder;
import net.minecraft.core.registries.Registries;
import net.minecraft.data.PackOutput;
import net.neoforged.neoforge.common.data.DatapackBuiltinEntriesProvider;
import net.neoforged.neoforge.registries.NeoForgeRegistries;
import redcrafter07.processed.ProcessedMod;
import redcrafter07.processed.worldgen.ModBiomeModifiers;
import redcrafter07.processed.worldgen.ModConfiguredFeatures;
import redcrafter07.processed.worldgen.ModPlacedFeatures;

import java.util.Set;
import java.util.concurrent.CompletableFuture;

class ModWorldGenProvider extends DatapackBuiltinEntriesProvider {
    private static final RegistrySetBuilder BUILDER = new RegistrySetBuilder()
            .add(Registries.CONFIGURED_FEATURE, ModConfiguredFeatures::bootstrap)
            .add(Registries.PLACED_FEATURE, ModPlacedFeatures::bootstrap)
            .add(NeoForgeRegistries.Keys.BIOME_MODIFIERS, ModBiomeModifiers::bootstrap);

    public ModWorldGenProvider(PackOutput output, CompletableFuture<HolderLookup.Provider> registries) {
        super(output, registries, BUILDER, Set.of(ProcessedMod.ID));
    }
}

File: src/main/kotlin/redcrafter07/processed/datagen/ModWorldGenProvider.kt
package redcrafter07.processed.datagen

import net.minecraft.core.HolderLookup
import net.minecraft.core.RegistrySetBuilder
import net.minecraft.core.registries.Registries
import net.minecraft.data.PackOutput
import net.neoforged.neoforge.common.data.DatapackBuiltinEntriesProvider
import net.neoforged.neoforge.registries.NeoForgeRegistries
import redcrafter07.processed.ProcessedMod
import redcrafter07.processed.worldgen.ModBiomeModifiers
import redcrafter07.processed.worldgen.ModConfiguredFeatures
import redcrafter07.processed.worldgen.ModPlacedFeatures
import java.util.concurrent.CompletableFuture

internal class ModWorldGenProvider(output: PackOutput, registries: CompletableFuture<HolderLookup.Provider?>) :
    DatapackBuiltinEntriesProvider(output, registries, BUILDER, setOf(ProcessedMod.ID)) {
    companion object {
        private val BUILDER: RegistrySetBuilder =
            RegistrySetBuilder().add(Registries.CONFIGURED_FEATURE, ModConfiguredFeatures::bootstrap)
                .add(Registries.PLACED_FEATURE, ModPlacedFeatures::bootstrap)
                .add(NeoForgeRegistries.Keys.BIOME_MODIFIERS, ModBiomeModifiers::bootstrap)
    }
}